### PR TITLE
rename MeanAveragePrecision

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

unify names in detection package, part of #729 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/detection/map.py [hunk 1]
```diff
@@ -731,3 +735,74 @@ def compute(self) -> dict:
...
+
+
+class MAP(MeanAveragePrecision):
+    r"""
+    Computes the `Mean-Average-Precision (mAP) and Mean-Average-Recall (mAR)
+    <https://jonathan-hui.medium.com/map-mean-average-precision-for-object-detection-45c121a31173>`_
+    for object detection predictions.
+
+    .. deprecated:: v0.7
+        Use :class:`torchmetrics.detect.MeanAveragePrecision`. Will be removed in v0.8.
```
Comment (rohitgr7) [inline torchmetrics/detection/map.py:747]:
```suggestion
        Use :class:`torchmetrics.detection.MeanAveragePrecision`. Will be removed in v0.8.
```

File: torchmetrics/detection/map.py [hunk 2]
```diff
@@ -731,3 +735,74 @@ def compute(self) -> dict:
         metrics.map_per_class = map_per_class_values
         metrics[f"mar_{self.max_detection_thresholds[-1]}_per_class"] = mar_max_dets_per_class_values
         return metrics
+
+
+class MAP(MeanAveragePrecision):
+    r"""
+    Computes the `Mean-Average-Precision (mAP) and Mean-Average-Recall (mAR)
+    <https://jonathan-hui.medium.com/map-mean-average-precision-for-object-detection-45c121a31173>`_
+    for object detection predictions.
```
Comment (rohitgr7) [inline torchmetrics/detection/map.py:744]:
```suggestion
    for object detection tasks.
```

File: torchmetrics/detection/map.py [hunk 3]
```diff
@@ -731,3 +735,74 @@ def compute(self) -> dict:
...
+        ...     labels=torch.IntTensor([0]),
+        ...   )
+        ... ]
+        >>> target = [
+        ...   dict(
+        ...     boxes=torch.Tensor([[214.0, 41.0, 562.0, 285.0]]),
+        ...     labels=torch.IntTensor([0]),
+        ...   )
+        ... ]
+        >>> metric = MAP()  # doctest: +SKIP
```
Comment (rohitgr7) [inline torchmetrics/detection/map.py:764]: I think import is missing for doctest.

**Review by rohitgr7 — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

File: torchmetrics/detection/map.py (hunk 3 above)
Comment (Borda) [reply torchmetrics/detection/map.py:764]: I do not want to show a user an import for deprecated function but still want to keep this doctest as validation that it still works as expected... :rabbit:

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**